### PR TITLE
Add SuperOffice provider

### PR DESCRIPTION
**Pull request by SuperDJ**

### CODE REVIEW

File: src/SuperOffice/Provider.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	/**
+	 * @param string $token
+	 *
+	 * @return array
+	 * @throws \GuzzleHttp\Exception\GuzzleException
+	 */
+	protected function getUserByToken($token): array
+	{
+		$response = $this->getHttpClient()->get(
+			'https://'.$this->config['environment'].'.superoffice.com/'.$this->config['customer_id'].'/api/v1/User/currentPrincipal',
```
Comment (atymic) [inline src/SuperOffice/Provider.php:64]: could we use sprintf here instead please?

File: src/SuperOffice/Provider.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+				$state
+			);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function getTokenUrl(): string
+	{
+		return 'https://'.($this->config['environment'] ?: 'sod').'.superoffice.com/login/common/oauth/tokens';
```
Comment (atymic) [inline src/SuperOffice/Provider.php:42]: Also sprintf here would be a little cleaner (and above)

File: src/SuperOffice/Provider.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	 * @return array
+	 */
+	public static function additionalConfigKeys(): array
+	{
+		return [
+			'environment',
+			'customer_id'
+		];
+	}
+}
```
Comment (atymic) [inline src/SuperOffice/Provider.php:97]: Missing newline at the end of the file

File: src/SuperOffice/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Microsoft
+
+```bash
+composer require socialiteproviders/microsoft
```
Comment (atymic) [inline src/SuperOffice/README.md:4]: need to change this

File: src/SuperOffice/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Microsoft
```
Comment (atymic) [inline src/SuperOffice/README.md:1]: And this

File: src/SuperOffice/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+];
+```
+
+### Usage
+
+You should now be able to use the provider like you would regularly use Socialite (assuming you have the facade installed):
+
+```php
+return Socialite::driver('superoffice')->redirect();
+```
```
Comment (atymic) [inline src/SuperOffice/README.md:44]: Could you add a list of returned user fields and link to the docs?

Ex: https://socialiteproviders.com/Kakao/#returned-user-fields

File: src/SuperOffice/composer.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+		"php": "^7.2",
+		"ext-json": "*",
+		"socialiteproviders/manager": "^3.6|^4.0"
+	},
+	"autoload": {
+		"psr-4": {
+			"SocialiteProviders\\SuperOffice\\": ""
+		}
+	}
+}
```
Comment (atymic) [inline src/SuperOffice/composer.json:21]: newline

**Review by atymic — CHANGES_REQUESTED** [review]
Looks good, just some minor stuff.

**Comment by SuperDJ** [comment]
@atymic what can be done to speed up the merge process?

**Comment by atymic** [comment]
Sorry, didn't see your last commit. Can you fix the styleci issues, then i'll merge? Thanks!

**Comment by SuperDJ** [comment]
@atymic that should do

**Review by atymic — APPROVED** [review]

**Pull request merged**